pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -0,0 +1,340 @@
+			"Operator managed resources.", err)
+		return 0
+	}
+	ormClient.cacheLock.Lock()
+	defer ormClient.cacheLock.Unlock()
+	// Clear existing cached operatorResourceSpecMap data
+	ormClient.operatorResourceSpecMap = make(map[string]*ORMSpec)
+	for _, ormCR := range ormCRs {
+		// An orm CR name always refers to a Operator manged CRD name
+		crdName := ormCR.GetName()
+		namespace := ormCR.GetNamespace()
+		// Get Operator CRs of the CRD in the same namespace of ORM CR.
+		operatorCRs, gvRes, err := ormClient.getOperatorCRsFromCRD(crdName, namespace)
+		if err != nil {
+			glog.Errorf("Failed to get Operator managed CRs from CRD %s, %v", crdName, err)
+			continue
+		}
+		ormTemplateMap, err := ormClient.populateORMTemplateMap(ormCR)
+		if err != nil {
+			glog.Errorf("Failed to populate ormTemplateMap for CRD %s: %v", crdName, err)
+			continue
+		}
+		for _, operatorCR := range operatorCRs {
+			crUID := string(operatorCR.GetUID())
+			operatorResourceSpec, exists := ormClient.operatorResourceSpecMap[crUID]
+			if exists {
+				glog.Errorf("Skip the duplicate CR UID '%s' for CRD %s", crUID, crdName)
+				continue
+			} else {
+				operatorResourceSpec = &ORMSpec{
+					operatorGVResource: gvRes,
+					ormTemplateMap:     ormTemplateMap,
+				}
+				ormClient.operatorResourceSpecMap[crUID] = operatorResourceSpec
+			}
+		}
+	}
+	return len(ormClient.operatorResourceSpecMap)
+}
+
+func (ormClient *ORMClient) getORMCRList() ([]unstructured.Unstructured, error) {
+	groupVersionRes := schema.GroupVersionResource{
+		Group:    ormGroup,
+		Version:  ormVersion,
+		Resource: ormResource,
+	}
+	ormCRs, err := ormClient.dynClient.Resource(groupVersionRes).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get OperatorResourceMapping CRs: %v", err)
+	}
+	return ormCRs.Items, nil
+}
+
+func (ormClient *ORMClient) getOperatorCRsFromCRD(crdName, namespace string) ([]unstructured.Unstructured, schema.GroupVersionResource, error) {
+	crds := ormClient.apiExtClient.CustomResourceDefinitions()
+	crd, err := crds.Get(crdName, metav1.GetOptions{})
+	if err != nil {
+		return nil, schema.GroupVersionResource{}, err
+	}
+	groupVersionRes := schema.GroupVersionResource{
+		Group:    crd.Spec.Group,
+		Version:  crd.Spec.Versions[0].Name,
+		Resource: crd.Spec.Names.Plural,
+	}
+	crs, err := ormClient.dynClient.Resource(groupVersionRes).Namespace(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, groupVersionRes, err
+	}
+	return crs.Items, groupVersionRes, nil
+}
+
+func (ormClient *ORMClient) populateORMTemplateMap(ormCR unstructured.Unstructured) (map[string]ORMTemplate, error) {
+	ormCRName := ormCR.GetName()
+	ormTemplateMap := make(map[string]ORMTemplate)
+	resourceMappings, _, err := unstructured.NestedSlice(ormCR.Object, ormResourceMappingsPath...)
+	if err != nil {
+		return nil, fmt.Errorf("no data under '%s' in ORM CR %s: %v", util.JSONPath(ormResourceMappingsPath), ormCRName, err)
```

Comment:
This should be based on the `bool` value return from the `NestedSlice` call. The flag tells if the field was found and `err` is generic error (which rarely happens on the methods on unstructured). 
Effectively if a field is not found, err would mostly be still nil.

Reply:
Ack!